solr/core/src/java/org/apache/solr/cloud/ZkController.java

```diff
@@ -2306,12 +2306,12 @@ private ZkCoreNodeProps waitForLeaderToSeeDownState(
         }
 
         // short timeouts, we may be in a storm and this is just best effort, and maybe we should be
-        // the
-        // leader now
+        // the leader now
+        // TODO .withHttpClient(getCoreContainer().getDefaultHttpSolrClient())
         try (SolrClient client =
-            new Builder(leaderBaseUrl)
+            new Http2SolrClient.Builder(leaderBaseUrl)
                 .withConnectionTimeout(8000, TimeUnit.MILLISECONDS)
-                .withSocketTimeout(30000, TimeUnit.MILLISECONDS)
+                .withIdleTimeout(30000, TimeUnit.MILLISECONDS)
                 .build()) {
```

Comment:
What about the security listeners?

Reply:
I changed it now.  It's a trade-off but at least the listener issue could be a real bug